### PR TITLE
Remove unnecessary repartition in TF2 ray estimator evaluate

### PR DESCRIPTION
**Pull request by hkvision**
Since we will assign multiple partitions for each worker, and combine partitions into a single Dataset in the actor, repartition is not needed. @yangw1234 Is it correct?

### CODE REVIEW

**Comment by yangw1234** [comment]
Does it work if the number of partitions is less the worker number?

**Comment by yangw1234** [comment]
> Does it work if the number of partitions is less the worker number?

Maybe add a unit test.

**Comment by hkvision** [comment]
> Does it work if the number of partitions is less the worker number?

For DataFrame when to SparkXShards will do repartition if partition is smaller. Maybe for SparkXShards need directly do this if the partition is smaller.

**Comment by hkvision** [comment]
Previously, for DataFrame input, when converting to SparkXShards, if the partition_num < num_workers, would do repartition.
But for direct SparkXShards input, if the original partition_num < num_workers, will get an error. So in this case we  also need to do repartition only if partition_num < num_workers.

Add unit tests to cover different cases.

File: python/orca/test/bigdl/orca/learn/ray/tf/test_tf_ray_estimator.py [hunk 1]
```diff
@@ -307,7 +307,7 @@ def test_horovod_learning_rate_schedule(self):
     def test_sparkxshards(self):
 
         train_data_shard = XShards.partition({"x": np.random.randn(100, 1),
-                                              "y": np.random.randint(0, 1, size=(100))})
+                                              "y": np.random.randint(0, 1, size=(100))}).repartition(8)
```
Comment (hkvision) [inline python/orca/test/bigdl/orca/learn/ray/tf/test_tf_ray_estimator.py:310]: To make sure data partition > num workers.

File: python/orca/test/bigdl/orca/learn/ray/tf/test_tf_ray_estimator.py [hunk 2]
```diff
@@ -321,10 +321,48 @@ def test_sparkxshards(self):
...
+        trainer.evaluate(df, batch_size=4, num_steps=25, feature_cols=["feature"],
+                         label_cols=["label"])
+        trainer.predict(df, feature_cols=["feature"]).collect()
+        trainer.shutdown()
+
     def test_dataframe(self):
 
         sc = init_nncontext()
-        rdd = sc.range(0, 10)
+        rdd = sc.range(0, 100).repartition(9)
```
Comment (hkvision) [inline python/orca/test/bigdl/orca/learn/ray/tf/test_tf_ray_estimator.py:365]: This tests for the case that partitions can't be evenly distributed by workers.

**Pull request merged**